include/NotificationsManager.php

```diff
@@ -28,109 +32,762 @@ public function __construct() {
-     * @return bool true on success, false on errors
-     */
-    public function setSeen($note, $seen = true) {
-        return q("UPDATE `notify` SET `seen` = %d WHERE ( `link` = '%s' OR ( `parent` != 0 AND `parent` = %d AND `otype` = '%s' )) AND `uid` = %d",
-            intval($seen),
-            dbesc($note['link']),
-            intval($note['parent']),
-            dbesc($note['otype']),
-            intval(local_user())
-        );
-    }
-       
-    /**
-     * @brief set seen state of all notifications of local_user()
-     *
-     * @param bool $seen optional true or false. default true
-     * @return bool true on success, false on error
-     */
-    public function setAllSeen($seen = true) {
-    	return q("UPDATE `notify` SET `seen` = %d WHERE `uid` = %d",
-            intval($seen),
+
+		if(dbm::is_result($r))
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+	/**
+	 * @brief Get network notifications
+	 * 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @param int $start Start the query at this point
+	 * @param int $limit Maximum number of query results
+	 * 
+	 * @return array with
+	 *	string 'ident' => Notification identifier
+	 *	int 'total' => Total number of available network notifications
+	 *	array 'notifications' => Network notifications
+	 */
+	public function networkNotifs($seen = 0, $start = 0, $limit = 80) {
+		$ident = 'network';
+		$total = $this->networkTotal($seen);
+		$notifs = array();
+
+		if($seen === 0)
+			$sql_seen = " AND `item`.`unseen` = 1 ";
+
+
+		$r = q("SELECT `item`.`id`,`item`.`parent`, `item`.`verb`, `item`.`author-name`, `item`.`unseen`,
+				`item`.`author-link`, `item`.`author-avatar`, `item`.`created`, `item`.`object` AS `object`,
+				`pitem`.`author-name` AS `pname`, `pitem`.`author-link` AS `plink`, `pitem`.`guid` AS `pguid`
+			FROM `item` INNER JOIN `item` AS `pitem` ON `pitem`.`id`=`item`.`parent`
+			WHERE `item`.`visible` = 1 AND `pitem`.`parent` != 0 AND
+				 `item`.`deleted` = 0 AND `item`.`uid` = %d AND `item`.`wall` = 0
+				$sql_seen
+			ORDER BY `item`.`created` DESC LIMIT %d, %d ",
+				intval(local_user()),
+				intval($start),
+				intval($limit)
+		);
+
+		if(dbm::is_result($r))
+			$notifs = $this->formatNotifs($r, $ident);
+
+		$arr = array (
+			'notifications' => $notifs,
+			'ident' => $ident,
+			'total' => $total,
+		);
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Total number of system notifications 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @return int Number of system notifications
+	 */
+	private function systemTotal($seen = 0) {
+		if($seen === 0)
+			$sql_seen = " AND `seen` = 0 ";
+
+		$r = q("SELECT COUNT(*) AS `total` FROM `notify` WHERE `uid` = %d $sql_seen",
 			intval(local_user())
 		);
-    }
+
+		if(dbm::is_result($r))
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+		/**
```

Comment:
there is a TAB to much ;-) just to have something to nitpick for now... and you to see that someone has looked at the PR